lib/compress/huf_compress.c

```diff
@@ -69,7 +69,7 @@ static size_t HUF_compressWeights (void* dst, size_t dstSize, const void* weight
     U32 tableLog = MAX_FSE_TABLELOG_FOR_HUFF_HEADER;
 
     FSE_CTable CTable[FSE_CTABLE_SIZE_U32(MAX_FSE_TABLELOG_FOR_HUFF_HEADER, HUF_TABLELOG_MAX)];
-    BYTE scratchBuffer[FSE_BUILD_CTABLE_WORKSPACE_SIZE(HUF_TABLELOG_MAX, MAX_FSE_TABLELOG_FOR_HUFF_HEADER)];
+    U32 scratchBuffer[FSE_BUILD_CTABLE_WORKSPACE_SIZE_U32(HUF_TABLELOG_MAX, MAX_FSE_TABLELOG_FOR_HUFF_HEADER)];
```

Comment:
minor question:

I'm struggling to understand how this could be an issue _only_ for 32-bit systems ...

Reply:
I believe that either compilers or the x86-64 ABI always align all stack entries to 8 or 16 bytes. But 32-bit systems wont.

Reply:
That's a good guess.
It's in line with observations that the bug disappears on enabling sanitizers.

Reply:
s390x also failed and is a 64bit arch. So probably more related to the alignment standards applied by compilers for the arch.